lib/Command/ExpirePassword.php

```diff
@@ -106,54 +129,128 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+
+		if ($allUsers !== false) {
+			$users = [];
+			$function = function (IUser $user) use (&$users) {
+				$users[] = $user;
+			};
+
+			$this->userManager->callForAllUsers($function);
+			$this->setPasswordExpiryForUsers($input, $output, $users);
+		} elseif (\count($groups) >= 1) {
+			$users = [];
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === false) {
+					$output->writeln("Group name $group doesn't exist");
```

Comment:
this doesn't feel right... you provide A, B, C and D, but B doesn't exists. Why A is processed but C and D not?
The user isn't warned about this behaviour. I might expect that all the existing groups are processed, or it might be acceptable if the command aborts if one group is missing, but abort the command half way seems wrong.

Reply:
The latest version of PR takes care of this issue.